### DISCUSSION TITLE
Distinguish .zim files that are not working with Kiwix

### DISCUSSION THREAD

**Discussion by Utopiah** [kiwix/kiwix-desktop#940] [Ideas]
I tried few .zim files without problem, e.g Wikipedia or StackOverflow, both via the desktop client and its server.

Unfortunately few files, e.g

- MDN https://library.kiwix.org/viewer#developer.mozilla.org_en_all_2023-02
- RPi docs https://library.kiwix.org/viewer#rapsberry_pi_docs_2023-01
- WordNet https://library.kiwix.org/viewer#wordnet_en_all_2023-04
- Off the grid https://library.kiwix.org/viewer#100r-off-the-grid_en_2022-06

can't be displayed due to the service worker error, which is well known e.g

- https://github.com/openzim/warc2zim/issues/73
- https://github.com/openzim/warc2zim/issues/98
- https://github.com/kiwix/overview/issues/69
- https://github.com/openzim/zimit/issues/145

Some of the solutions make sense but IMHO not available to the typical user, e.g sharing content behind a reverse proxy providing https. 

I would argue the file themselves are correct but the requirement at least here for kiwix-desktop make them non working. Consequently promoting them via the library means creating a lot of confusion for users who are not technically savvy.

Finally my recommendation here, to be pragmatic, is to hide by default such files from the library until, hopefully soon, this viewer actually supports them without the need for user customization.

PS: amazing work, thanks to Kiwix and OpenZIM, just learned about the project while reading https://stackoverflow.blog/2022/10/20/introducing-the-overflow-offline-project/ and spending a week offline while still being able to code!

**Comment by kelson42** [comment]
We already have hidden such files in the online library ([in 2.3.1 version](https://github.com/kiwix/kiwix-desktop/issues/780)) and we will add the [full support](https://github.com/kiwix/kiwix-desktop/issues/487) anyway soon. Should be in worse case in 2.5.0/. For the meantime you can read the files using the server feature (see menu entry "local  Kiwix Server" and then use IP `127.0.0.1`)... this is not more than 3 clicks.
